ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/RequiredPluginsInitializer.java

```diff
@@ -35,26 +41,64 @@ public class RequiredPluginsInitializer extends ClasspathContainerInitializer {
 				}
 			});
 
+	private static final DeferredClasspathContainerInitializerJob deferredClasspathContainerInitializerJob = new DeferredClasspathContainerInitializerJob();
+
+	private static class DeferredClasspathContainerInitializerJob extends Job {
+
+		private final Set<IJavaProject> projects = new LinkedHashSet<>();
+
+		public DeferredClasspathContainerInitializerJob() {
+			// This name is not displayed to a user.
+			super("DeferredClasspathContainerInitializerJob"); //$NON-NLS-1$
+			setSystem(true);
+		}
+
+		public synchronized void initialize(IJavaProject project) {
+			if (projects.add(project)) {
+				schedule();
+			}
+		}
+
+		private synchronized IJavaProject[] consumeProjects() {
+			try {
+				return projects.toArray(IJavaProject[]::new);
+			} finally {
+				projects.clear();
+			}
+		}
+
+		@Override
+		protected IStatus run(IProgressMonitor monitor) {
+			try {
+				setupClasspath(consumeProjects());
+			} catch (JavaModelException e) {
+				PDECore.log(e);
+			}
+			return Status.OK_STATUS;
+		}
+
+		@Override
+		public boolean belongsTo(Object family) {
+			return family == PluginModelManager.class;
+		}
+	}
+
 	@Override
 	public void initialize(IPath containerPath, IJavaProject javaProject) throws CoreException {
 		if (Job.getJobManager().isSuspended()) {
 			// if the jobmanager is currently suspended we can't use the
 			// schedule/join pattern here, instead we must retry the requested
 			// action once jobs are enabled again, this will the possibly
 			// trigger a rebuild if required or notify other listeners.
-			Job job = Job.create(PDECoreMessages.PluginModelManager_InitializingPluginModels, m -> {
-				setupClasspath(javaProject);
-			});
-			job.setSystem(true);
-			job.schedule();
+			deferredClasspathContainerInitializerJob.initialize(javaProject);
 		} else {
 			setupClasspath(javaProject);
 		}
 	}
 
-	protected void setupClasspath(IJavaProject javaProject) throws JavaModelException {
-		IProject project = javaProject.getProject();
-		// The first project to be built may initialize the PDE models, potentially long running, so allow cancellation
+	protected static void setupClasspath(IJavaProject... javaProjects) throws JavaModelException {
```

Comment:
Is it really useful to change this method for the case we actually like to avoid? How many projects are there to be collected that its worth to optimize here?